### PR TITLE
feat: introduce golangci

### PR DESCRIPTION
**Pull request by kushthedude**
Signed-off-by: Kush Trivedi <kushthedude@gmail.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,95 +9,25 @@ on:
...
-       with:
-         version: "v0.7.0"
-     - run: |
-         export CURRENTCONTEXT="$(kubectl config current-context)"
-         echo "current-context:" ${CURRENTCONTEXT}   
-         export KUBECONFIG="${HOME}/.kube/config"
-         echo "environment-kubeconfig:" ${KUBECONFIG}
-         GOPROXY=direct GOSUMDB=off GO111MODULE=on go test ./...
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.29
```
Comment (mgfeller) [inline .github/workflows/ci.yml:21]: latest is already 1.31

Comment (leecalcote) [reply .github/workflows/ci.yml:21]: @kushthedude ready to upgrade?

**Comment by kushthedude** [comment]
Nope, some viper config error new to me. May be @kumarabd can figure the
issue out?

**Comment by leecalcote** [comment]
@kushthedude is this valid anymore of is it superseded by https://github.com/layer5io/meshery-kuma/pull/107?

**Comment by leecalcote** [comment]
> @kushthedude is this valid anymore of is it superseded by #107?

And given #109?

**Comment by kushthedude** [comment]
@leecalcote let me merge upstream master into the branch and see if golangci works

**Pull request closed without merging**